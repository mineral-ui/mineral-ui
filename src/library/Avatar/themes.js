/* @flow */
import type { AvatarThemeFn } from './types';

export const avatarTheme: AvatarThemeFn = (baseTheme) => ({
  Avatar_fontSize_small: baseTheme.fontSize_mouse,
  Avatar_fontSize_medium: baseTheme.fontSize_ui,
  Avatar_fontSize_large: baseTheme.h4_fontSize,
  Avatar_fontSize_jumbo: baseTheme.h4_fontSize,
  Avatar_fontWeight: baseTheme.fontWeight_bold,
  Avatar_size_small: baseTheme.size_small,
  Avatar_size_medium: baseTheme.size_medium,
  Avatar_size_large: baseTheme.size_large,
  Avatar_size_jumbo: baseTheme.size_jumbo,

  ...baseTheme
});
