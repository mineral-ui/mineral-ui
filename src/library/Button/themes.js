/* @flow */
import { pxToEm } from '../styles';

import type { ButtonThemeFn } from './types';

// prettier-ignore
export const buttonTheme: ButtonThemeFn = (baseTheme) => ({
  Button_backgroundColor: baseTheme.backgroundColor,
  Button_backgroundColor_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_focus: baseTheme.backgroundColor_focus,
  Button_backgroundColor_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_minimal_active: baseTheme.backgroundColor_active,
  Button_backgroundColor_minimal_hover: baseTheme.backgroundColor_hover,
  Button_backgroundColor_primary: baseTheme.backgroundColor_themePrimary,
  Button_backgroundColor_primary_active: baseTheme.backgroundColor_themePrimary_active,
  Button_backgroundColor_primary_focus: baseTheme.backgroundColor_themePrimary_focus,
  Button_backgroundColor_primary_hover: baseTheme.backgroundColor_themePrimary_hover,
  Button_borderColor: baseTheme.borderColor,
  Button_borderColor_active: baseTheme.borderColor_theme_active,
  Button_borderColor_focus: baseTheme.borderColor_theme_focus,
  Button_borderColor_hover: baseTheme.borderColor_theme_hover,
  Button_borderRadius: baseTheme.borderRadius_3,
  Button_borderWidth: 1, // px,
  Button_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${baseTheme.borderColor_theme_focus}`,
  Button_color: baseTheme.color_theme,
  Button_color_minimal: baseTheme.color_theme,
  Button_color_primary: baseTheme.color_themePrimary,
  Button_fontWeight: baseTheme.fontWeight_semiBold,
  Button_paddingHorizontal: baseTheme.space_inset_sm_plus,
  Button_paddingIconOnly_small: pxToEm(3),
  Button_paddingIconOnly_medium: pxToEm(7),
  Button_paddingIconOnly_large: pxToEm(7),
  Button_paddingIconOnly_jumbo: pxToEm(13),
  Button_height_small: baseTheme.size_small,
  Button_height_medium: baseTheme.button_size_medium,
  Button_height_large: baseTheme.button_size_large,
  Button_height_jumbo: baseTheme.button_size_jumbo,

  ButtonContent_fontSize: baseTheme.fontSize_ui_sm,
  ButtonContent_fontSize_small: pxToEm(12),

  ButtonIcon_color: baseTheme.icon_color_theme,
  ButtonIcon_margin: baseTheme.space_inset_sm,

  ...baseTheme
});
