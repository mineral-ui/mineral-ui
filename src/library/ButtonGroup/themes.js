/* @flow */
import type { ButtonGroupThemeFn } from './types';

export const buttonGroupTheme: ButtonGroupThemeFn = (baseTheme) => ({
  ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
  ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
  ButtonGroupButton_borderStartColor: baseTheme.borderColor,
  ButtonGroupButton_borderStartColor_checked: 'currentcolor',
  ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,

  ...baseTheme
});
