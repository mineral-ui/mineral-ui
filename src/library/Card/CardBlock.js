/* @flow */
import React from 'react';
import { createStyledComponent } from '../styles';
import { componentTheme as cardRowComponentTheme } from './CardRow';
import CardRow from './CardRow';

type Props = {
  /** Contents of CardBlock */
  children: React$Node
};

export const componentTheme = (baseTheme: Object) => ({
  CardBlock_fontSize: baseTheme.fontSize_ui,
  CardBlock_lineHeight: baseTheme.lineHeight_prose,

  ...baseTheme
});

const styles = {
  inner: ({ theme: baseTheme }) => {
    const theme = componentTheme(baseTheme);

    return {
      fontSize: theme.CardBlock_fontSize,
      lineHeight: theme.CardBlock_lineHeight
    };
  },
  root: ({ theme: baseTheme }) => {
    const theme = cardRowComponentTheme(baseTheme);

    return {
      '&:last-child': {
        marginBottom: theme.CardRow_marginVerticalLast
      }
    };
  }
};

const Inner = createStyledComponent('div', styles.inner);
const Root = createStyledComponent(CardRow, styles.root, {
  displayName: 'CardBlock'
});

/**
 * CardBlock is used to normalize font sizes for content and to provide
 * consistent margins and padding.
 */
export default function CardBlock(props: Props) {
  const { children, ...restProps } = props;
  return (
    <Root {...restProps}>
      <Inner>{children}</Inner>
    </Root>
  );
}
