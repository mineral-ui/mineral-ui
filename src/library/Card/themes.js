/* @flow */
import { pxToEm } from '../styles';

import type {
  CardThemeFn,
  CardActionsThemeFn,
  CardBlockThemeFn,
  CardDividerThemeFn,
  CardFooterThemeFn,
  CardStatusThemeFn,
  CardTitleThemeFn
} from './types';

// prettier-ignore
export const cardTheme: CardThemeFn = (baseTheme) => ({
  Card_backgroundColor: baseTheme.panel_backgroundColor,
  Card_borderColor: baseTheme.panel_borderColor,
  Card_borderRadius: baseTheme.borderRadius_1,
  Card_boxShadow: baseTheme.boxShadow_2,
  Card_boxShadow_focus: `0 0 0 2px ${baseTheme.borderColor_theme_focus}, ${baseTheme.boxShadow_1}`,

  CardRow_marginVertical: baseTheme.space_stack_md,
  CardRow_marginVerticalLast: baseTheme.space_stack_lg,
  CardRow_paddingHorizontal: baseTheme.space_inset_md,

  ...baseTheme
});

export const cardActionsTheme: CardActionsThemeFn = (baseTheme) => ({
  CardActionsAction_spaceInline: baseTheme.space_inline_sm,

  ...baseTheme
});

export const cardBlockTheme: CardBlockThemeFn = (baseTheme) => ({
  CardBlock_fontSize: baseTheme.fontSize_ui,
  CardBlock_lineHeight: baseTheme.lineHeight_prose,

  ...baseTheme
});

export const cardDividerTheme: CardDividerThemeFn = (baseTheme) => ({
  CardDivider_borderColor: baseTheme.borderColor,
  CardDivider_borderWidth: '1px',

  ...baseTheme
});

export const cardFooterTheme: CardFooterThemeFn = (baseTheme) => ({
  CardFooter_backgroundColor: baseTheme.well_backgroundColor,
  CardFooter_borderColor: baseTheme.borderColor,

  CardFooterRow_marginVertical: baseTheme.space_stack_sm,
  CardFooterRow_marginVerticalLast: baseTheme.space_stack_md,

  CardFooterTitle_color: baseTheme.h5_color,
  CardFooterTitle_fontSize: baseTheme.h5_fontSize,
  CardFooterTitle_fontWeight: baseTheme.h5_fontWeight,

  ...baseTheme
});

export const cardStatusTheme: CardStatusThemeFn = (baseTheme) => ({
  CardStatus_fontSize: baseTheme.fontSize_ui,
  CardStatus_fontWeight: baseTheme.fontWeight_regular,

  CardStatusIcon_margin: baseTheme.space_inline_sm,
  CardStatusIcon_size: pxToEm(12),

  ...baseTheme
});

export const cardTitleTheme: CardTitleThemeFn = (baseTheme) => ({
  CardTitle_color: baseTheme.h4_color,
  CardTitle_fontSize: baseTheme.h4_fontSize,
  CardTitle_fontWeight: baseTheme.h4_fontWeight,

  CardTitleAvatar_margin: baseTheme.space_inline_sm,
  CardTitleAvatarSize: baseTheme.size_small,
  CardTitleAvatarSize_large: baseTheme.size_large,

  CardTitleIcon_margin: baseTheme.space_inline_sm,

  CardTitleSecondaryText_color: baseTheme.color_mouse,
  CardTitleSecondaryText_fontSize: baseTheme.fontSize_mouse,
  CardTitleSecondaryText_fontWeight: baseTheme.fontWeight_regular,

  CardSubtitle_color: baseTheme.color_mouse,
  CardSubtitle_fontSize: baseTheme.fontSize_mouse,
  CardSubtitle_fontWeight: baseTheme.fontWeight_regular,
  CardSubtitle_marginTop: baseTheme.space_stack_sm,

  ...baseTheme
});
