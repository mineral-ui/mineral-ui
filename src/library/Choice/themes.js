/* @flow */
import { pxToEm } from '../styles';

import type { ChoiceThemeFn, ChoiceGroupThemeFn } from './types';

export const choiceTheme: ChoiceThemeFn = (baseTheme) => {
  const colors = {
    background: baseTheme.input_backgroundColor,
    regular: baseTheme.borderColor,
    checked: baseTheme.borderColor_theme,
    focus: baseTheme.borderColor_theme_focus,
    hover: baseTheme.borderColor_theme_hover
  };

  // prettier-ignore
  return {
    ChoiceControl_backgroundColor: colors.background,
    ChoiceControl_backgroundColor_checked: colors.checked,
    ChoiceControl_backgroundColor_checkedHover: colors.hover,
    ChoiceControl_borderColor: colors.regular,
    ChoiceControl_borderColor_hover: colors.hover,
    ChoiceControl_borderColor_checked: colors.checked,
    ChoiceControl_borderColor_checkedHover: colors.hover,
    ChoiceControl_borderRadius: baseTheme.borderRadius_1,
    ChoiceControl_boxShadow_focus: `0 0 0 1px ${baseTheme.boxShadow_focusInner}, 0 0 0 2px ${colors.focus}`,
    ChoiceControl_size: pxToEm(16),
    ChoiceControl_size_jumbo: pxToEm(24),

    ChoiceText_color: baseTheme.color,
    ChoiceText_fontSize: baseTheme.fontSize_ui,
    ChoiceText_fontSize_small: pxToEm(12),
    ChoiceText_marginHorizontal: baseTheme.space_inline_md,

    ...baseTheme
  };
};

export const choiceGroupTheme: ChoiceGroupThemeFn = (baseTheme) => ({
  ChoiceGroupControl_marginHorizontal_inline: baseTheme.space_inline_xl,
  ChoiceGroupControl_marginVertical_stacked: baseTheme.space_stack_md,
  ChoiceGroupControl_marginVertical_stackedJumbo: baseTheme.space_stack_lg,

  ...baseTheme
});
