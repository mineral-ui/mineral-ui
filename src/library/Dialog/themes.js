/* @flow */
import { pxToEm } from '../styles';
import { mapComponentThemes } from '../themes';
import { overflowContainerWithShadowsTheme } from '../OverflowContainer/themes';

import type {
  DialogThemeFn,
  DialogActionsThemeFn,
  DialogBodyThemeFn,
  DialogRowThemeFn,
  DialogTitleThemeFn
} from './types';

export const dialogTheme: DialogThemeFn = (baseTheme) => ({
  Dialog_transitionDuration: '250ms',
  Dialog_zIndex: baseTheme.zIndex_1600,

  DialogCloseButton_margin: baseTheme.space_inline_sm,

  DialogContent_backgroundColor: baseTheme.panel_backgroundColor,
  DialogContent_borderColor: baseTheme.panel_borderColor,
  DialogContent_borderRadius: baseTheme.borderRadius_1,
  DialogContent_boxShadow: baseTheme.boxShadow_5,
  DialogContent_maxHeight: '80vh',
  DialogContent_maxHeight_small: pxToEm(560),
  DialogContent_maxHeight_medium: pxToEm(560),
  DialogContent_maxHeight_large: pxToEm(720),
  DialogContent_maxWidth_small: pxToEm(400),
  DialogContent_maxWidth_medium: pxToEm(640),
  DialogContent_maxWidth_large: pxToEm(1200),
  DialogContent_minWidth: pxToEm(360),
  DialogContent_offsetVertical: baseTheme.space_stack_xxl,
  DialogContent_width_small: '35vw',
  DialogContent_width_medium: '50vw',
  DialogContent_width_large: '80vw',

  DialogOverlay_backgroundColor: 'rgba(0, 0, 0, 0.6)',

  ...dialogRowTheme(baseTheme),
  ...baseTheme
});

export const dialogActionsTheme: DialogActionsThemeFn = (baseTheme) => ({
  DialogActionsItem_margin: baseTheme.space_stack_sm,

  ...baseTheme
});

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const dialogBodyTheme: DialogBodyThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'OverflowContainerWithShadows',
      theme: overflowContainerWithShadowsTheme(baseTheme)
    },
    {
      name: 'DialogBody',
      theme: {}
    },
    baseTheme
  );

export const dialogRowTheme: DialogRowThemeFn = (baseTheme) => ({
  DialogRow_fontSize: baseTheme.fontSize_ui,
  DialogRow_paddingHorizontal: baseTheme.space_inline_lg,
  DialogRow_marginVertical: baseTheme.space_stack_lg,

  ...baseTheme
});

export const dialogTitleTheme: DialogTitleThemeFn = (baseTheme) => ({
  DialogTitle_color: baseTheme.h4_color,
  DialogTitle_fontSize: baseTheme.h4_fontSize,
  DialogTitle_fontWeight: baseTheme.h4_fontWeight,

  DialogTitleIcon_margin: baseTheme.space_inline_sm,
  DialogTitleIcon_size: pxToEm(24),

  ...baseTheme
});
