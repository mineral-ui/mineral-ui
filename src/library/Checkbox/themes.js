/* @flow */
import { mapComponentThemes } from '../themes';
import { choiceTheme, choiceGroupTheme } from '../Choice';

import type { CheckboxThemeFn, CheckboxGroupThemeFn } from './types';

import { pxToEm } from '../styles';

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxTheme: CheckboxThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'Choice',
      theme: choiceTheme(baseTheme)
    },
    {
      name: 'Checkbox',
      theme: {
        CheckboxControl_backgroundColor_checked: baseTheme.checkbox_backgroundColor_checked,
        CheckboxControl_backgroundColor_checkedHover: baseTheme.checkbox_backgroundColor_checkedHover,
        CheckboxControl_borderColor: baseTheme.checkbox_borderColor,
        CheckboxControl_borderColor_checked: baseTheme.checkbox_borderColor_checked,
        CheckboxControl_borderColor_checkedHover: baseTheme.checkbox_borderColor_checkedHover,
        CheckboxControl_borderColor_hover: baseTheme.checkbox_borderColor_hover,
        CheckboxControl_borderRadius: baseTheme.borderRadius_2,
        CheckboxControl_borderWidth: `${baseTheme.checkbox_borderWidth}px`,
        CheckboxControl_size: pxToEm(20),
        CheckboxText_marginHorizontal: baseTheme.space_inline_sm
      }
    },
    baseTheme
  );

// $FlowFixMe - strict theme keys incompatible with mapComponentThemes
export const checkboxGroupTheme: CheckboxGroupThemeFn = (baseTheme) =>
  mapComponentThemes(
    {
      name: 'ChoiceGroup',
      theme: choiceGroupTheme(baseTheme)
    },
    {
      name: 'CheckboxGroup',
      theme: {}
    },
    baseTheme
  );
