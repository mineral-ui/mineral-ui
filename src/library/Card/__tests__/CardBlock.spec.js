/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import CardBlock, { componentTheme } from '../../Card/CardBlock';
import testThemeOverrides from '../../../../utils/testThemeOverrides';
import { getProcessedComponentThemeKeys } from '../../themes/processComponentTheme';

function shallowCardBlock(props = {}) {
  const cardBlockProps = {
    children: 'children',
    ...props
  };
  return shallow(<CardBlock {...cardBlockProps} />);
}

describe('CardBlock', () => {
  it('renders', () => {
    const cardBlock = shallowCardBlock();

    expect(cardBlock.exists()).toEqual(true);
  });

  describe('theme overrides', () => {
    testThemeOverrides(
      <CardBlock>children</CardBlock>,
      getProcessedComponentThemeKeys(componentTheme)
    );
  });
});
