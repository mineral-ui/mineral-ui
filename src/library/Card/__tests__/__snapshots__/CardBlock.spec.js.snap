// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBlock demo examples Snapshots: children 1`] = `
.emotion-9 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
}

.emotion-9:last-child {
  margin-bottom: 1.5em;
}

.emotion-8 {
  font-size: 0.875em;
  line-height: 1.5;
}

.emotion-11::after {
  clear: both;
  content: "";
  display: table;
}

.emotion-11 > * {
  margin-bottom: 1em;
}

@media (min-width:40em) {
  .emotion-11 > * {
    float: left;
    margin-right: 1em;
    width: 47%;
  }
}

@media (min-width:64em) {
  .emotion-11 > *:nth-last-child(n+3),
  .emotion-11 > *:nth-last-child(n+3) ~ * {
    width: 30%;
  }
}

.emotion-10 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: #ffffff;
  border: 1px solid #ebeff5;
  border-radius: 0.1875em;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2),0 4px 8px 0 rgba(0,0,0,0.2);
  padding-bottom: 0.01em;
  padding-top: 0.01em;
}

.emotion-10 *,
.emotion-10 *::before,
.emotion-10 *::after {
  box-sizing: inherit;
}

.emotion-10:focus {
  box-shadow: 0 0 0 2px #1d5bbf,0 1px 2px 0 rgba(0,0,0,0.2),0 2px 4px 0 rgba(0,0,0,0.2);
}

.emotion-3 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-1 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 > [role="img"] {
  margin-right: 0.5em;
  position: relative;
  top: 0.25em;
}

.emotion-0 {
  color: #58606e;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.125em;
  font-weight: 700;
  margin: 0;
}

.emotion-7 {
  background-color: #ebeff5;
  padding: 1em 0;
}

.emotion-6 {
  box-sizing: border-box;
  color: #3272d9;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: #ffffff;
  border-color: #c8d1e0;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: inline-block;
  font-weight: 600;
  height: 2.5em;
  margin: 0;
  min-width: 2.5em;
  padding: 0 0.5em;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  width: 100%;
}

.emotion-6 *,
.emotion-6 *::before,
.emotion-6 *::after {
  box-sizing: inherit;
}

.emotion-6:focus {
  background-color: #ffffff;
  box-shadow: 0 0 0 1px #ffffff,0 0 0 2px #1d5bbf;
  color: #3272d9;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:hover {
  background-color: #f5f7fa;
  border-color: #5691f0;
  color: #3272d9;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:focus:active,
.emotion-6:focus:hover {
  border-color: #c8d1e0;
}

.emotion-6:active {
  background-color: #ebeff5;
  border-color: #1d5bbf;
  color: #3272d9;
}

.emotion-6::-moz-focus-inner {
  border: 0;
}

.emotion-6 [role="img"] {
  box-sizing: content-box;
  color: #3272d9;
  display: block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-6 [role="img"]:first-child {
  margin-right: 0.5em;
}

.emotion-6 [role="img"]:last-child {
  margin-left: 0.5em;
}

.emotion-6 [role="img"]:only-child {
  margin: 0;
}

.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-height: 100%;
  pointer-events: none;
  width: 100%;
}

.emotion-4 {
  display: block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  font-size: 0.875em;
  line-height: 2.857142857142857em;
}

.emotion-4:first-child {
  padding-left: 0.5714285714285714em;
}

.emotion-4:last-child {
  padding-right: 0.5714285714285714em;
}

<Styled(div)>
  <div
    className="emotion-11"
  >
    <Card>
      <Card>
        <div
          className="emotion-10"
        >
          <CardTitle>
            <CardTitle>
              <div
                className="emotion-3"
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    <Styled(div)>
                      <div
                        className="emotion-1"
                      >
                        <Styled(h3)>
                          <h3
                            className="emotion-0"
                          >
                            Card Title
                          </h3>
                        </Styled(h3)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </CardTitle>
          </CardTitle>
          <CardBlock>
            <CardBlock>
              <div
                className="emotion-9"
              >
                <Styled(div)>
                  <div
                    className="emotion-8"
                  >
                    <Styled(div)>
                      <div
                        className="emotion-7"
                      >
                        <Button
                          element="button"
                          fullWidth={true}
                          size="large"
                          type="button"
                        >
                          <Button
                            element="button"
                            fullWidth={true}
                            size="large"
                            text="Button"
                            type="button"
                          >
                            <button
                              className="emotion-6"
                              type="button"
                            >
                              <Styled(span)>
                                <span
                                  className="emotion-5"
                                >
                                  <Styled(span)
                                    size="large"
                                  >
                                    <span
                                      className="emotion-4"
                                    >
                                      Button
                                    </span>
                                  </Styled(span)>
                                </span>
                              </Styled(span)>
                            </button>
                          </Button>
                        </Button>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </CardBlock>
          </CardBlock>
        </div>
      </Card>
    </Card>
  </div>
</Styled(div)>
`;

exports[`CardBlock demo examples Snapshots: consistent-spacing 1`] = `
.emotion-4 {
  font-size: 0.875em;
  line-height: 1.5;
}

.emotion-9::after {
  clear: both;
  content: "";
  display: table;
}

.emotion-9 > * {
  margin-bottom: 1em;
}

@media (min-width:40em) {
  .emotion-9 > * {
    float: left;
    margin-right: 1em;
    width: 47%;
  }
}

@media (min-width:64em) {
  .emotion-9 > *:nth-last-child(n+3),
  .emotion-9 > *:nth-last-child(n+3) ~ * {
    width: 30%;
  }
}

.emotion-8 {
  box-sizing: border-box;
  color: #333840;
  font-family: "Open Sans",-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  -moz-osx-font-smoothing: auto;
  -webkit-font-smoothing: antialiased;
  background-color: #ffffff;
  border: 1px solid #ebeff5;
  border-radius: 0.1875em;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2),0 4px 8px 0 rgba(0,0,0,0.2);
  padding-bottom: 0.01em;
  padding-top: 0.01em;
}

.emotion-8 *,
.emotion-8 *::before,
.emotion-8 *::after {
  box-sizing: inherit;
}

.emotion-8:focus {
  box-shadow: 0 0 0 2px #1d5bbf,0 1px 2px 0 rgba(0,0,0,0.2),0 2px 4px 0 rgba(0,0,0,0.2);
}

.emotion-3 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-1 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 > [role="img"] {
  margin-right: 0.5em;
  position: relative;
  top: 0.25em;
}

.emotion-0 {
  color: #58606e;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.125em;
  font-weight: 700;
  margin: 0;
}

.emotion-6 {
  position: relative;
}

.emotion-6::before {
  border-color: #f0f5fc;
  border-style: solid;
  border-width: 1em 1em 1.5em;
  bottom: -1.5em;
  content: "";
  left: 0;
  position: absolute;
  right: 0;
  top: -1em;
}

.emotion-5 {
  margin-bottom: 1em;
  margin-top: 1em;
  padding-left: 1em;
  padding-right: 1em;
  position: relative;
}

.emotion-5:last-child {
  margin-bottom: 1.5em;
}

.emotion-5::before {
  border-color: #f0f5fc;
  border-style: solid;
  border-width: 1em 1em 1.5em;
  bottom: -1.5em;
  content: "";
  left: 0;
  position: absolute;
  right: 0;
  top: -1em;
}

<Styled(div)>
  <div
    className="emotion-9"
  >
    <Card>
      <Card>
        <div
          className="emotion-8"
        >
          <CardTitle>
            <CardTitle>
              <div
                className="emotion-3"
              >
                <Styled(div)>
                  <div
                    className="emotion-2"
                  >
                    <Styled(div)>
                      <div
                        className="emotion-1"
                      >
                        <Styled(h3)>
                          <h3
                            className="emotion-0"
                          >
                            Card Title
                          </h3>
                        </Styled(h3)>
                      </div>
                    </Styled(div)>
                  </div>
                </Styled(div)>
              </div>
            </CardTitle>
          </CardTitle>
          <Styled(CardBlock)>
            <CardBlock
              className="emotion-6"
            >
              <CardBlock
                className="emotion-6"
              >
                <div
                  className="emotion-5"
                >
                  <Styled(div)>
                    <div
                      className="emotion-4"
                    >
                      
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec mattis
  pretium massa. Aliquam erat volutpat. Nulla facilisi.

                    </div>
                  </Styled(div)>
                </div>
              </CardBlock>
            </CardBlock>
          </Styled(CardBlock)>
        </div>
      </Card>
    </Card>
  </div>
</Styled(div)>
`;
